src/OAuth2/Controller/TokenController.php

```diff
@@ -97,9 +97,15 @@ public function handleTokenRequest(RequestInterface $request, ResponseInterface
      */
     public function grantAccessToken(RequestInterface $request, ResponseInterface $response)
     {
+        if (strtolower($request->server('REQUEST_METHOD')) == 'options') {
+            $response->addHttpHeaders(array('Allow' => 'POST, OPTIONS'));
+
+            return null;
+        }
+        
         if (strtolower($request->server('REQUEST_METHOD')) != 'post') {
             $response->setError(405, 'invalid_request', 'The request method must be POST when requesting an access token', '#section-3.2');
-            $response->addHttpHeaders(array('Allow' => 'POST'));
```

Comment:
what is the reason of deletion this line? it should be here. Everything else looks good.